openedx/core/djangoapps/theming/helpers.py

```diff
@@ -288,6 +288,18 @@ def get_theme_base_dirs():
     if not is_comprehensive_theming_enabled():
         return []
 
+    # Legacy code for COMPREHENSIVE_THEME_DIR backward compatibility
+    if hasattr(settings, "COMPREHENSIVE_THEME_DIR"):
+        theme_dir = settings.COMPREHENSIVE_THEME_DIR
+
+        if not isinstance(theme_dir, basestring):
+            raise ImproperlyConfigured("COMPREHENSIVE_THEME_DIRS must be a string.")
```

Comment:
If this block remains after considering the above comment, needs to be renamed to "COMPREHENSIVE_THEME_DIR".